File: support_tools/swigwin-3.0.12/Examples/test-suite/octave/inherit_missing_runme.m
inherit_missing

a = inherit_missing.new_Foo();
b = inherit_missing.Bar();
c = inherit_missing.Spam();

x = inherit_missing.do_blah(a);
if (!strcmp(x, "Foo::blah"))
    error("Whoa! Bad return %s", x)
endif

x = inherit_missing.do_blah(b);
if (!strcmp(x, "Bar::blah"))
    error("Whoa! Bad return %s", x)
endif

x = inherit_missing.do_blah(c);
if (!strcmp(x, "Spam::blah"))
    error("Whoa! Bad return %s", x)
endif

inherit_missing.delete_Foo(a);

File: support_tools/swigwin-3.0.12/Examples/test-suite/perl5/inherit_missing_runme.pl
#!/usr/bin/perl
use strict;
use warnings;
use Test::More tests => 5;
BEGIN { use_ok('inherit_missing') }
require_ok('inherit_missing');

# adapted from ../python/inherit_missing_runme.py

my $a = inherit_missing::new_Foo();
my $b = inherit_missing::Bar->new();
my $c = inherit_missing::Spam->new();

is(inherit_missing::do_blah($a), "Foo::blah");

is(inherit_missing::do_blah($b), "Bar::blah");

is(inherit_missing::do_blah($c), "Spam::blah");

inherit_missing::delete_Foo($a);
